src/chttpd/src/chttpd_misc.erl

```diff
@@ -334,9 +329,33 @@ handle_reload_query_servers_req(#httpd{method='POST'}=Req) ->
 handle_reload_query_servers_req(Req) ->
     send_method_not_allowed(Req, "POST").
 
+handle_uuids_req(#httpd{method='GET'}=Req) ->
+    Max = list_to_integer(config:get("uuids","max_count","1000")),
+    Count = try list_to_integer(couch_httpd:qs_value(Req, "count", "1")) of
+        N when N > Max ->
+            throw({bad_request, <<"count parameter too large">>});
+        N when N < 0 ->
+            throw({bad_request, <<"count must be a positive integer">>});
+        N -> N
+    catch
+        error:badarg ->
+            throw({bad_request, <<"count must be a positive integer">>})
+    end,
+    UUIDs = [couch_uuids:new() || _ <- lists:seq(1, Count)],
+    Etag = couch_httpd:make_etag(UUIDs),
+    couch_httpd:etag_respond(Req, Etag, fun() ->
+        CacheBustingHeaders = [
+            {"Date", couch_util:rfc1123_date()},
+            {"Cache-Control", "no-cache"},
+            % Past date, ON PURPOSE!
+            {"Expires", "Mon, 01 Jan 1990 00:00:00 GMT"},
+            {"Pragma", "no-cache"},
+            {"ETag", Etag}
```

Comment:
I think it's to ensure the responses are not cached. This was a verbatim copy and paste from https://github.com/apache/couchdb/blob/403d27b16f5209f918766dc62a8b0d039080fd32/src/couch/src/couch_httpd_misc_handlers.erl#L96-L108

Reply:
If we think an `ETag` is necessary for some compatibility, should we maybe just generate something random here to avoid the hashing computation overhead?

Reply:
Good idea. Maybe we can just use the first 1st uuid since we already did the work to generate them. Something like?:

```erlang
Etag = binary_to_list(hd(UUIDs))
``` 

But let's make it a separate PR. As this one is already quite large :-)

Reply:
Sounds good!